src/Symfony/Component/Profiler/Profiler.php

```diff
@@ -0,0 +1,387 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Profiler;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollectorInterface as DeprecatedDataCollectorInterface;
+use Symfony\Component\HttpKernel\DataCollector\LateDataCollectorInterface as DeprecatedLateDataCollectorInterface;
+use Symfony\Component\Profiler\DataCollector\DataCollectorInterface;
+use Symfony\Component\Profiler\DataCollector\RuntimeDataCollectorInterface;
+use Symfony\Component\Profiler\DataCollector\LateDataCollectorInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Profiler\Storage\ProfilerStorageInterface;
+
+/**
+ * Profiler.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Jelte Steijaert <jelte@khepri.be>
+ */
+class Profiler
+{
+    /**
+     * @var DataCollectorInterface[]
+     */
+    private $collectors = array();
+
+    /**
+     * @var ProfilerStorageInterface
+     */
+    private $storage;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var bool
+     */
+    private $enabled = true;
+
+    /**
+     * Constructor.
+     *
+     * @param ProfilerStorageInterface $storage A ProfilerStorageInterface instance
+     * @param LoggerInterface          $logger  A LoggerInterface instance
+     */
+    public function __construct(ProfilerStorageInterface $storage, LoggerInterface $logger = null)
+    {
+        $this->storage = $storage;
+        $this->logger = $logger;
+    }
+
+    public function generateToken()
+    {
+        return substr(hash('sha256', uniqid(mt_rand(), true)), 0, 6);
+    }
+
+    /**
+     * Disables the profiler.
+     */
+    public function disable()
+    {
+        $this->enabled = false;
+    }
+
+    /**
+     * Enables the profiler.
+     */
+    public function enable()
+    {
+        $this->enabled = true;
+    }
+
+    /**
+     * Loads the Profile for the given token.
+     *
+     * @param string $token A token
+     *
+     * @return ProfileInterface A Profile instance
+     */
+    public function load($token)
```

Comment:
I don't disagree and the only reason to keep it might be **DX**, but using the `Storage` shouldn't be a big step.

but the storage will need to be publicly accessible then
```xml
<service id="profiler.storage" class="" public="false">
```